src/Microsoft.AspNet.Mvc.Core/Rendering/Html/DefaultDisplayTemplates.cs

```diff
@@ -25,12 +26,12 @@ public static string BooleanTemplate(IHtmlHelper htmlHelper)
 
             return htmlHelper.ViewData.ModelMetadata.IsNullableValueType ?
                 BooleanTemplateDropDownList(htmlHelper, value) :
-                BooleanTemplateCheckbox(value ?? false);
+                BooleanTemplateCheckbox(value ?? false, htmlHelper.HtmlEncoder);
         }
 
-        private static string BooleanTemplateCheckbox(bool value)
+        private static string BooleanTemplateCheckbox(bool value, IHtmlEncoder htmlEncoder)
```

Comment:
Looked at this in more detail and it would be a pretty substantial refactor. Might we worth considering at some point but not as part of this change.